### PR TITLE
add BEFORE_UPLOAD collection state

### PR DESCRIPTION
**Pull request by eldrickb**
Adds the BEFORE_UPLOAD collection state
- implements `<vox-re-record-button />` and `<vox-playback-button />`
- cleans up `RecordButton`

preview: 
image

### CODE REVIEW

**Comment by mmallett** [comment]
Does the listen button work? I'd like to see a demo

File: ui/src/components/PlaybackButton.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    constructor() {
+        super();
+
+        this.audio = undefined;
+        this.playing = false;
+    }
+
+    updated() {
+        // only update if audio isn't set
+        if (!this.audio && this.audioUrl) {
```
Comment (mmallett) [inline ui/src/components/PlaybackButton.js:46]: audio url is a window object url? the one we get from the record button?

File: ui/src/components/PlaybackButton.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+            this.audio = new Audio(this.audioUrl);
+        }
+    }
+
+    handleClick() {
```
Comment (mmallett) [inline ui/src/components/PlaybackButton.js:51]: pretty slick

File: ui/src/components/RecordButton.js [hunk 3]
```diff
@@ -40,116 +43,138 @@ export class RecordButton extends LitElement {
...
-            if (audio.recordingUrl) {
-                try {
-                    const resp = await this.utteranceService.saveAudio(audio);
 
-                    if (!resp) throw new Error();
-                } catch (e) {
-                    // If upload failed, pivot to upload error collection state
-                    this.dispatchCollectionState(CollectionStates.UPLOAD_ERROR);
-                }
+                if (!resp) throw new Error();
```
Comment (mmallett) [inline ui/src/components/RecordButton.js:131]: nit, add a message to the error constructor

File: ui/src/components/PlaybackButton.js (hunk 2 above)
Comment (ashleygranitto) [inline ui/src/components/PlaybackButton.js:51]: add documentation

File: ui/src/components/ReRecordButton.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+export class ReRecordButton extends LitElement {
+    static get properties() {
+        return {};
+    }
+
+    static get styles() {
+        return style;
+    }
+
+    dispatchReRecord() {
```
Comment (ashleygranitto) [inline ui/src/components/ReRecordButton.js:32]: add documentation

File: ui/src/components/RecordButton.js [hunk 5]
```diff
@@ -40,116 +43,138 @@ export class RecordButton extends LitElement {
...
     constructor() {
         super();
+        this.disabled = false;
         this.audioRecorder = new AudioRecorder();
         this.utteranceService = new UtteranceApiService();
+        this.finishedAudio = undefined;
     }
 
-    updated() {
+    updated(changedProperties) {
```
Comment (ashleygranitto) [inline ui/src/components/RecordButton.js:52]: nit: do you need to include changedProperties if you're calling the functions regardless of the properties changed?

File: ui/src/components/RecordingSection.js [hunk 6]
```diff
@@ -34,30 +39,53 @@ export class RecordingSection extends LitElement {
...
+            },
+            bubbles: true,
+            composed: true,
+        });
+
+        this.dispatchEvent(event);
+    }
+
+    /**
+     * Rendering related
```
Comment (ashleygranitto) [inline ui/src/components/RecordingSection.js:68]: nit: fix spacing, make documentation a bit more specific

File: ui/src/styles/components/PlaybackButton.css.js [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {css} from 'lit-element';
+
+export default css`
+    button {
+        display: flex;
```
Comment (ashleygranitto) [inline ui/src/styles/components/PlaybackButton.css.js:21]: nit: alphabetical order

**Pull request merged**